lxd/container_lxc.go

```diff
@@ -5299,24 +5315,20 @@ func (c *containerLXC) fillNetworkDevice(name string, m shared.Device) (shared.D
 				return nil, err
 			}
 
-			c.localConfig[configKey] = volatileName
-			c.expandedConfig[configKey] = volatileName
-
 			// Update the database
-			tx, err := dbBegin(c.daemon.db)
-			if err != nil {
-				return nil, err
-			}
-
-			err = dbContainerConfigInsert(tx, c.id, map[string]string{configKey: volatileName})
+			err = updateKey(configKey, volatileName)
 			if err != nil {
-				tx.Rollback()
-				return nil, err
-			}
+				// Check if something else filled it in behind our back
+				value, err1 := dbContainerConfigGet(c.daemon.db, c.id, configKey)
+				if err1 != nil || value == "" {
+					return nil, err
```

Comment:
Doesn't this leave us in a state where the hw addr can get updated, but the name won't? Before we rolled both updates back, but now we don't.